### PR TITLE
feat: scale up lower decimal oracle price

### PR DESCRIPTION
**Pull request by chad-js**

### CODE REVIEW

File: src/ChainlinkTrigger.sol
Comment (chad-js) [inline src/ChainlinkTrigger.sol:83]: @Dendrimer and I went back and forth a bit on how best to implement this feature. The current implementation matches @Dendrimer's suggestion of storing a single scale factor and an enum value to determine which oracle price to apply it to.

My original approach was to save a scale factor for each oracle and then always apply both in `programmaticCheck`. So if the decimals are the same, we read two factors from storage and multiply (they'd both be `1`). If there's a diff, one gets multiplied by some scale factor and the other by `1`.

@davidlaprade @mds1 lmk thoughts on which you prefer. I think the current solution is probably more intuitive but adds more lines. Also if the decimals are the same (which I think is true for most cases), we save an SLOAD in `programmaticCheck`.

File: src/ChainlinkTrigger.sol
Comment (mds1) [inline src/ChainlinkTrigger.sol]: Do we need rpow here? Seems `_scaleFactor = 10 ** (_truthOracleDecimals - _trackingOracleDecimals)` is equivalent

Comment (mds1) [inline src/ChainlinkTrigger.sol]: Do we need rpow here? Seems `_scaleFactor = 10 ** (_truthOracleDecimals - _trackingOracleDecimals)` is equivalent

File: src/ChainlinkTrigger.sol
Comment (mds1) [reply src/ChainlinkTrigger.sol:83]: Since all the scale factors and which one to scale are immutables, there will be no SLOADs! 

Given that, I would go with whichever we think is more clear/readable, since both work equally well and gas costs should be similar enough given that all new params involved are immutables. I think branchless (store both / multiply both) might be _slightly_ cheaper (and IMO branchless is kinda nice, just feels clean) cause `JUMPI` costs 10 gas and `MUL` costs 5 gas, but that is negligible.

So I'll leave it up to you guys 🙂

File: test/ChainlinkTrigger.t.sol
Comment (mds1) [inline test/ChainlinkTrigger.t.sol:117]: Some (admittedly rare / unlikely to occur in practice) edge cases I'd recommend testing:
- One oracle has 0 decimals, other has e.g. 18
- One oracle has 1 decimal, other has e.g. 18
- One oracle has > 18 decimals, other has e.g. 18

**Review by mds1 — COMMENTED** [review]
LGTM overall though!

File: src/ChainlinkTrigger.sol
Comment (mds1) [reply src/ChainlinkTrigger.sol]: (I think this comment got duplicated from my wifi? not deleting any just in case it's only actually rendering twice for me currently)
image

Comment (chad-js) [reply src/ChainlinkTrigger.sol]: ah I thought `rpow` is cheaper, but ok ill switch

File: test/ChainlinkTrigger.t.sol
Comment (chad-js) [reply test/ChainlinkTrigger.t.sol:117]: added!

File: src/ChainlinkTrigger.sol
Comment (chad-js) [reply src/ChainlinkTrigger.sol]: updateed

File: src/ChainlinkTrigger.sol
Comment (mds1) [reply src/ChainlinkTrigger.sol:83]: Leaving this unresolved for @davidlaprade to see it — would be good if we can get a review from him since he did most of the chainlink work originally IIRC

**Review by mds1 — APPROVED** [review]
LGTM! (also there's conflict to resolve)

**Review by Dendrimer — APPROVED** [review]

**Pull request merged**
